benedict/dicts/base/base_dict.py

```diff
@@ -65,7 +65,6 @@ def __repr__(self):
     def __setitem__(self, key, value):
         if self._pointer:
             self._dict[key] = value
-            return
```

Comment:
This will keep the data in two places though.

Reply:
Is this an issue? I see that all the unit tests have passed. 

And to make [JsonSerializable via subclassing dict, this definitely requires your data to be in the superclass](https://stackoverflow.com/a/31207881/80353)

Unless there's a different approach to solve `json.dumps(d)` or drop the support for `json.dumps(d)`

Reply:
I cannot tell how big of an issue this is. The way the `pointer`-ed benedict and the other is architected, I'd say that it is intended to be wrapped, rather than keeping a copy around. But I am not familiar with the codebase, so I am not sure.

Regarding `json.dumps`, you are right that it's not possible without this. But rather than trying to make `json.dumps(benedict_obj)` work, I'd prefer `benedict_obj.json()`. That way, we can just roll back e8ccb7270 and fix #59.

Reply:
> But rather than trying to make json.dumps(benedict_obj) work, I'd prefer benedict_obj.json()

In that case, the documentation needs to be clear about two things:

1. that benedict is NOT JsonSerializable and therefore `json.dumps(benedict_obj)` does not work and is intended to be so.
2. that keeping only one set of the same data is a non-functional requirement and is intended to be so. Not sure if this can be made into a unit test.

Reply:
Sorry for the late reply.

For me this is an old dilemma, the problem is that python `json` module internally uses `c_make_encoder` [under the hood](https://github.com/python/cpython/blob/3.9/Lib/json/encoder.py#L246) and it doesn't call any dunder method, so it is not possible to decide which `dict` should be passed to the encoder.

This is the state of things:
- I want to keep benedict 100% dict compatible
- I don't want to keep data on two different dicts, there will be surely some problem soon or later

Yes, I want both things, and since the problem is related to how the `json` module works, a possible *dirty* solution is to hack the `json` module, what do you think about it?

Reply:
I understand your desire to have both and respect that. In the perfect world, I also wnat that. Given we not in perfect world, you need to choose one over the other in the short term.

Then explore possible solutions for the long term to achieve the perfect world. That's all I have to say on this topic for now.